### PR TITLE
Chosen select fields

### PR DESCRIPTION
**Pull request by dalepotter**
~~Ensure #49 is merged first!~~ <- Done!

Improves select fields using jQuery chosen.

### CODE REVIEW

**Comment by andylolz** [comment]
It looks like I used https://github.com/alxlit/bootstrap-chosen to style this. A quick search shows there are forks of that that will work for bootstrap 4.

File: chosen.css [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+}
+
+.chosen-container-single .chosen-single abbr {
+  position: absolute;
+  top: 6px;
+  right: 26px;
+  display: block;
+  width: 12px;
+  height: 12px;
+  background: url("chosen-sprite.png") -42px 1px no-repeat;
```
Comment (andylolz) [inline chosen.css]: chosen-sprite.png is missing.

File: chosen.css [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+/* @group Retina compatibility */
+@media only screen and (-webkit-min-device-pixel-ratio: 1.5), only screen and (min-resolution: 144dpi), only screen and (min-resolution: 1.5dppx) {
+  .chosen-rtl .chosen-search input[type="text"],
+  .chosen-container-single .chosen-single abbr,
+  .chosen-container-single .chosen-single div b,
+  .chosen-container-single .chosen-search input[type="text"],
+  .chosen-container-multi .chosen-choices .search-choice .search-choice-close,
+  .chosen-container .chosen-results-scroll-down span,
+  .chosen-container .chosen-results-scroll-up span {
+    background-image: url("chosen-sprite@2x.png") !important;
```
Comment (andylolz) [inline chosen.css]: chosen-sprite@2x.png is also missing.

**Comment by andylolz** [comment]
Might be tidier / bit more conventional to put the static files in a /static directory, or /static/img etc. Or ideally manage them with [yarn](https://yarnpkg.com/lang/en/) or similar.

**Comment by dalepotter** [comment]
> Might be tidier / bit more conventional to put the static files in a /static directory, or /static/img etc. Or ideally manage them with yarn or similar.

Fixed in a2876a2

File: templates/index.html [hunk 3]
```diff
@@ -163,7 +165,8 @@ <h1 dir="ltr">IATI Data Store CSV Query Builder (Alpha Version)</h1>
             <label for="entry_18398991">
               <div>Select Type of Reporting Organisation (eg. INGO = 21)</div>
             </label>
-            <select class="form-control" multiple="multiple" name="entry.18398991[]" size="6" value="" id="entry_18398991[]">
+            <div class="clearfix"></div>
+            <select data-placeholder="Choose a reporting organisation type..." multiple class="chosen-select form-control" id="entry_18398991[]" name="entry.18398991[]">
```
Comment (allthatilk) [inline templates/index.html]: Inconsistent capitalisation

Comment (dalepotter) [reply templates/index.html]: Thanks, will fix...

Comment (dalepotter) [reply templates/index.html]: Fixed in 6f0988a

**Comment by allthatilk** [comment]
Looks good 👍

**Comment by andylolz** [comment]
It is chosen… But it looks a bit ugly because it’s not bootstrap4 compatible. Compare/contrast with #57!

**Comment by andylolz** [comment]
NB this fixes #34.

**Pull request merged**